test/Microsoft.AspNetCore.Routing.Tests/Template/TemplateMatcherTests.cs

```diff
@@ -991,6 +991,107 @@ public void TryMatch_MultipleOptionalParameters()
             Assert.False(values.ContainsKey("id"));
         }
 
+        [Theory]
+        [InlineData("///")]
+        [InlineData("/a//")]
+        [InlineData("/a/b//")]
+        [InlineData("//b//")]
+        [InlineData("///c")]
+        [InlineData("///c/")]
+        public void TryMatch_MultipleOptionalParameters_WithEmptyIntermediateSegmentsDoesNotMatch(string url)
```

Comment:
I do think otherwise, which is why I specifically mentioned both. We don't need a test for every possible scenario, just a few.